src/app/test-record/test-section/edit/test-section-edit.component.ts

```diff
@@ -0,0 +1,158 @@
+import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';
+import { TestType } from '@app/models/test.type';
+import { TestResultModel } from '@app/models/test-result.model';
+import { MatDialog } from '@angular/material/dialog';
+import { DialogBoxConfirmationComponent } from '@app/shared/dialog-box-confirmation/dialog-box-confirmation.component';
+import { Router } from '@angular/router';
+import {
+  REASON_FOR_ABANDONING_HGV_TRL,
+  REASON_FOR_ABANDONING_PSV,
+  REASON_FOR_ABANDONING_TIR,
+  RESULT
+} from '@app/test-record/test-record.enums';
+import {
+  ControlContainer,
+  FormControl,
+  FormGroup,
+  FormGroupDirective,
+  Validators
+} from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { DisplayOptionsPipe } from '@app/pipes/display-options.pipe';
+import { SelectOption } from '@app/models/select-option';
+import { TestRecordMapper } from '@app/test-record/test-record.mapper';
+
+@Component({
+  selector: 'vtm-test-section-edit',
+  templateUrl: './test-section-edit.component.html',
+  styleUrls: ['./test-section-edit.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  viewProviders: [{ provide: ControlContainer, useExisting: FormGroupDirective }]
+})
+export class TestSectionEditComponent implements OnInit {
+  @Input() testType: TestType;
+  @Input() testRecord: TestResultModel;
+  @Input() applicableTestTypeIds1: {};
+  @Input() applicableTestTypeIds2: {};
+  resultOptions: string[] = Object.values(RESULT);
+  testResultChildForm: FormGroupDirective;
+  testTypeGroup: FormGroup;
+  testResultSubscription: Subscription;
+  isAbandoned: boolean;
+  prohibitionOptionSelected: string;
+  prohibitionOptions: SelectOption[];
+  reasonsForAbandoning: string[];
+
+  constructor(
+    private dialog: MatDialog,
+    private router: Router,
+    parentForm: FormGroupDirective,
+    private testRecordMapper: TestRecordMapper
+  ) {
+    this.testResultChildForm = parentForm;
+  }
+
+  ngOnInit() {
+    this.isAbandoned = this.testType.testResult === 'abandoned';
```

Comment:
I think this could be moved into a service so that we remove any logic in there and make it testable